src/Canvas.cc

```diff
@@ -782,57 +773,57 @@ Canvas::GetWeightFromCSSString(const char *weight) {
   return w;
 }
 
+bool streq_casein(std::string& str1, std::string& str2) {
+  return str1.size() == str2.size() && std::equal(str1.begin(), str1.end(), str2.begin(), [](char& c1, char& c2) {
+    return c1 == c2 || std::toupper(c1) == std::toupper(c2);
+  });
+}
+
 /*
  * Given a user description, return a description that will select the
  * font either from the system or @font-face
  */
 
 PangoFontDescription *
 Canvas::ResolveFontDescription(const PangoFontDescription *desc) {
-  FontFace best;
-  PangoFontDescription *ret = NULL;
-
   // One of the user-specified families could map to multiple SFNT family names
   // if someone registered two different fonts under the same family name.
   // https://drafts.csswg.org/css-fonts-3/#font-style-matching
-  char **families = g_strsplit(pango_font_description_get_family(desc), ",", -1);
-  GHashTable *seen_families = g_hash_table_new(g_str_hash, g_str_equal);
-  GString *resolved_families = g_string_new("");
-
-  for (int i = 0; families[i]; ++i) {
-    GString *renamed_families = g_string_new("");
-    std::vector<FontFace>::iterator it = _font_face_list.begin();
-
-    for (; it != _font_face_list.end(); ++it) {
-      if (g_ascii_strcasecmp(families[i], pango_font_description_get_family(it->user_desc)) == 0) {
-        char *name = g_strdup(pango_font_description_get_family(it->sys_desc));
```

Comment:
And I think this was the other. `g_hash_table_destroy` doesn't do anything to free its contained data unless the table is created with `key/value_destroy_func`s.